packages/playground/src/components/project-settings/members-list.tsx

```diff
@@ -0,0 +1,183 @@
+  const [editMember, setEditMember] = useState<MembersProps | null>(null) // Store member being edited
+  const [deleteMember, setDeleteMember] = useState<MembersProps | null>(null) // Store member being deleted
+
+  //open delete dialog for a specific member
+  const openDeleteDialog = (member: MembersProps) => {
+    setDeleteMember(member)
+    setIsDialogDeleteOpen(true)
+  }
+
+  // Open the edit dialog for a specific member
+  const openEditDialog = (member: MembersProps) => {
+    setEditMember(member)
+    setIsDialogEditOpen(true)
+  }
+
+  const handleRoleSave = () => {
+    setIsDialogEditOpen(false) // Close dialog after save
+  }
+
+  const moreActionsTooltip = ({ member }: { member: MembersProps }) => {
+    return (
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button variant="ghost" size="xs">
+            <Icon name="vertical-ellipsis" size={14} className="text-tertiary-background cursor-pointer" />
+          </Button>
+        </DropdownMenuTrigger>
+        <DropdownMenuContent className="shadow-sm py-2 bg-primary-background border border-gray-800 rounded-[10px] w-[180px]">
+          <DropdownMenuGroup>
+            <DropdownMenuItem
+              className="cursor-pointer"
+              onSelect={() => {
+                openEditDialog(member)
+                setEditMember(member) // Set the clicked member's data
+                setIsDialogEditOpen(true) // Open the edit dialog
+              }}>
+              <DropdownMenuShortcut className="ml-0">
+                <Icon name="edit-pen" className="mr-2" />
+              </DropdownMenuShortcut>
+              Edit
+            </DropdownMenuItem>
+            <DropdownMenuSeparator />
+            <DropdownMenuItem
+              className="cursor-pointer text-red-400 hover:text-red-400 focus:text-red-400"
+              onSelect={() => {
+                openDeleteDialog(member)
+              }}>
+              <DropdownMenuShortcut className="ml-0">
+                <Icon name="trash" className="mr-2 text-red-400" />
+              </DropdownMenuShortcut>
+              Remove member
+            </DropdownMenuItem>
+          </DropdownMenuGroup>
+        </DropdownMenuContent>
+      </DropdownMenu>
+    )
+  }
+
+  return (
+    <>
+      <Table variant="asStackedList">
+        <TableHeader>
+          <TableRow>
+            <TableHead className="text-primary">Name</TableHead>
+            <TableHead className="text-primary">Role</TableHead>
+            <TableHead className="text-primary">Email</TableHead>
+            {members[0]?.timestamp && <TableHead className="text-right text-primary">Date added</TableHead>}
+            <TableHead>
+              <></>
+            </TableHead>
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {members &&
+            members.map((member, index) => {
+              return (
+                <TableRow key={index}>
```

Comment:
instead of index maybe use some other key?

Reply:
use `user.uid` instead